persistent/Database/Persist/Quasi.hs

```diff
@@ -413,11 +413,14 @@ Unfortunately, we can't use this to create Haddocks for you, because <https://gi
 -}
 module Database.Persist.Quasi
     ( parse
-    , PersistSettings (..)
+    , PersistSettings
     , upperCaseSettings
     , lowerCaseSettings
-    , toFKNameInfixed
+    , setPsUseSnakeCaseForiegnKeys
     , nullable
     ) where
 
 import Database.Persist.Quasi.Internal
+
+setPsUseSnakeCaseForiegnKeys :: PersistSettings -> PersistSettings
+setPsUseSnakeCaseForiegnKeys ps = ps { psToFKName = toFKNameInfixed "_" }
```

Comment:
Bit of a long winded function name, but I went with this approach because I think people will want to either have this switched on or off. There is room for further flexibility though if this is wanted, but maybe that would be overkill.

I also thought with this change, that `persistent` should have this behaviour as default, potentially allowing people to configure this if they wanted to (though why they would I'm not sure), though unfortunately if we did that off the bat then that would break existing persistent codebases, right? Not sure what the long term fix is here, or it might have to be something that came along with a major version bump. Even along with a major version bump we are making people deal with renaming their foreign keys when there is actually technically wrong with them (they're just not aesthetically pleasing to the human reader).

Another approach with a major bump (or minor, I'm not sure which of the two this would fit in), would be to swap this functionality, so rather than _opt-in_ it would become _opt-out_, if for some reason people would rather just change the value in `PersistentSettings` than changing things in the DB.

Again, just thinking out loud here :)

Reply:
IMO the default behavior should be consistent with whatever is currently happening - so end users wouldn't notice a change from this new option.

I'd prefer to see a more general setter exposed, even if this is the most likely variant that folks will want. People *can* just refer to the `.Internal` module but that's clunky.